### PR TITLE
Wrap in.. menu react select

### PR DESCRIPTION
**Pull request by balazsbajorics**
**Problem:**
I was foolish and I thought I can handroll react-select's basic functionality in a day

**Fix:**
Use react-select instead

**Commit Details:**
- I'm using a react-select
- The big react-select style object `componentSelectorStyles` is in a hook, memoized to the current theme.
- looks like a floating window, behaves like a react select

image

### CODE REVIEW

File: editor/src/components/canvas/ui/floating-insert-menu.tsx [hunk 1]
```diff
@@ -91,63 +92,147 @@ function useGetInsertableComponents(filterString: string): InsertableComponentFl
...
-    }, [insertableComponent, onMouseOver])
+function useComponentSelectorStyles(): StylesConfig {
+  const colorTheme = useColorTheme()
+  // componentSelectorStyles will only be recreated if the theme changes, otherwise we re-use the same object
+  return React.useMemo(
+    () => ({
+      container: (styles) => ({
+        // the outermost element. It contains the popup menu,  so don't set a height on it!
+        // shouldn't contain any sizing
+        // ...styles,
```
Comment (Rheeseyb) [inline editor/src/components/canvas/ui/floating-insert-menu.tsx:110]: There are a lot of `// ...styles`. Can you delete those please.

File: editor/src/components/canvas/ui/floating-insert-menu.tsx [hunk 2]
```diff
@@ -91,63 +92,147 @@ function useGetInsertableComponents(filterString: string): InsertableComponentFl
...
+        // default styles mess with layout, so ignore them
+        // ...styles,
+        display: 'flex',
+        position: 'relative',
+        flexGrow: 1,
+        flexShrink: 0,
+        overflowX: 'scroll',
+        alignItems: 'center',
+        gap: 4,
+        // height: 22,
```
Comment (Rheeseyb) [inline editor/src/components/canvas/ui/floating-insert-menu.tsx:137]:
```suggestion
```

File: editor/src/components/canvas/ui/floating-insert-menu.tsx [hunk 3]
```diff
@@ -191,116 +271,59 @@ export var FloatingMenu = () => {
...
+        </div>
+
+        <Select
+          autoFocus
+          isMulti={false}
+          controlShouldRenderValue={false}
+          hideSelectedOptions={false}
+          menuIsOpen
+          // eslint-disable-next-line react/jsx-no-bind
+          onChange={(value, action) => {
```
Comment (Rheeseyb) [inline editor/src/components/canvas/ui/floating-insert-menu.tsx:313]: `React.useCallback` please

Comment (balazsbajorics) [reply editor/src/components/canvas/ui/floating-insert-menu.tsx]: fixed

File: editor/src/components/canvas/ui/floating-insert-menu.tsx (hunk 2 above)
Comment (balazsbajorics) [reply editor/src/components/canvas/ui/floating-insert-menu.tsx]: fixed

**Review by Rheeseyb — APPROVED** [review]

File: editor/src/components/canvas/ui/floating-insert-menu.tsx (hunk 1 above)
Comment (Rheeseyb) [reply editor/src/components/canvas/ui/floating-insert-menu.tsx:110]: Discussed elsewhere. We've agreed to disagree on this, with the outcome being that this should be left to highlight that we are explicitly omitting the passed in `styles`.

**Review by seanparsons — APPROVED** [review]

**Pull request merged**